Sources/TuistGenerator/Linter/TargetLinter.swift

```diff
@@ -94,13 +94,13 @@ class TargetLinter: TargetLinting {
         var issues: [LintingIssue] = []
 
         let files = target.resources.map(\.path)
-        let infoPlists = files.filter { $0.pathString.contains("Info.plist") }
         let entitlements = files.filter { $0.pathString.contains(".entitlements") }
 
-        issues.append(contentsOf: infoPlists.map {
-            let reason = "Info.plist at path \($0.pathString) being copied into the target \(target.name) product."
-            return LintingIssue(reason: reason, severity: .warning)
-        })
+        if let targetInfoPlistPath = target.infoPlist?.path, files.contains(targetInfoPlistPath) {
```

Comment:
I'm inclined towards removing the check. Here's my why:
- This logic might need to traverse a large list of files. I don't think it'll be a lot if we compare the time that takes with the total generation time, but every bit of time we add aggregates to the other bits.
- This is a check that Xcode runs early in the process, so it's a bit redundant. Also I think that the error that Xcode surfaces is very obvious, so I don't think we are adding much value from our end.

Reply:
I'd be still inclined to keep it because you may write some globs which accidentally include your Info.plist file without knowing.

If we are concerned about performance we can use a `Set` which is `O(1)` for `contains` - but I really don't think that it's worth it right now.

Reply:
Let's leave it then. If we see this becomes a performance bottleneck, we can either move those attributes to sets, or consider running linters in parallel, considering that they are not doing permutations of the models.

Reply:
That's a good point. I didn't know about the Xcode warning since I've never tried to add the Info.plist to the resources. I'm fine with removing it 👍 

Should I just go ahead and do it or is it something that has to be discussed further? @olejnjak [mentioned on the issue](https://github.com/tuist/tuist/issues/1170#issuecomment-610896992) that he thinks the warning is quite nice.

Reply:
Ah, I didn't see your replies. Just forget my comment 😃